src/Lime.Transport.SignalR/EnvelopeHub.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Lime.Protocol.Network;
+using Lime.Protocol.Serialization;
+using Microsoft.AspNetCore.Http.Connections;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Lime.Transport.SignalR
+{
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Performance", "CA1812: Remove internal classes that are never instantiated", Justification = "The class is instantiated via reflection by ASP.NET")]
+    internal class EnvelopeHub : Hub
+    {
+        private readonly Channel<ITransport> _transportChannel;
+        private readonly ITraceWriter _traceWriter;
+        private readonly IEnvelopeSerializer _envelopeSerializer;
+        private readonly IHubContext<EnvelopeHub> _hubContext;
+        private readonly HubOptions _hubOptions;
+        private readonly HttpConnectionDispatcherOptions _httpConnectionDispatcherOptions;
+        private readonly EnvelopeHubOptions _envelopeHubOptions;
+        private readonly ConcurrentDictionary<string, Channel<string>> _clientChannels;
+
+        public EnvelopeHub(Channel<ITransport> transportChannel,
+                           ConcurrentDictionary<string, Channel<string>> clientChannels,
+                           IEnvelopeSerializer envelopeSerializer,
+                           IHubContext<EnvelopeHub> hubContext,
+                           HubOptions hubOptions,
+                           HttpConnectionDispatcherOptions httpConnectionDispatcherOptions,
+                           EnvelopeHubOptions envelopeHubOptions,
+                           ITraceWriter traceWriter = null)
+        {
+            _transportChannel = transportChannel;
+            _clientChannels = clientChannels;
+            _traceWriter = traceWriter;
+            _envelopeSerializer = envelopeSerializer;
+            _hubContext = hubContext;
+            _hubOptions = hubOptions;
+            _httpConnectionDispatcherOptions = httpConnectionDispatcherOptions;
+            _envelopeHubOptions = envelopeHubOptions;            
+        }
+
+        public async Task FromClient(string envelope)
+        {
+            if (_clientChannels.TryGetValue(Context.UserIdentifier, out var channel))
```

Comment:
It would be better if we do not compromise the `UserIdentifier` property for this. In the future, we may use to allow *Transport authentication* with SignalR.

Reply:
As discussed offline, we are limited as to how we can fetch an `IClientProxy`, having to use either `ConnectionId` or `UserIdentifier`. I'll keep this convo. open tho while I think about other possible solutions.